### PR TITLE
`toBeVisible` implies `toBeInTheDocument`

### PR DESCRIPTION
**Pull request by calebeby**
**What/Why**:

https://github.com/testing-library/jest-dom/issues/338

**Checklist**:

- [x] Documentation
- [x] Tests
- [ ] Updated Type Definitions - The `.d.ts` is not in this repo
- [x] Ready to be merged

cc @gnapse

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -429,6 +430,7 @@ An element is visible if **all** the following conditions are met:
   [`hidden`](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/hidden)
   attribute
 - if `<details />` it has the `open` attribute
+- it is present in the document
```
Comment (gnapse) [inline README.md:433]: nit pick: I'd put this one first.

File: src/__tests__/to-be-visible.js [hunk 2]
```diff
@@ -242,4 +243,11 @@ describe('.toBeVisible', () => {
       })
     })
   })
+  describe('element not in document', () => {
```
Comment (gnapse) [inline src/__tests__/to-be-visible.js:246]: a couple of nit-picks:

```suggestion
  })

  describe('when element not in document', () => {
```

Although my real recommendation here would be to put this test case more towards the beginning, as a sibling case of the first `it` block. I feel there's no need for this to be inside a `describe` block of its own.

**Review by gnapse — APPROVED** [review]
Thanks for this!

Left a couple of nit-picky comments. Let me know if/when you address them and we're good to go.

**Comment by calebeby** [comment]
@gnapse Should be all set!

**Review by gnapse — APPROVED** [review]

**Comment by gnapse** [comment]
@all-contributors please add @calebeby for code, test

**Pull request merged**